src/west/commands/project.py

```diff
@@ -208,6 +193,49 @@ def do_run(self, args, user_args):
             _git(project, 'status', extra_args=user_args)
 
 
+class Update(WestCommand):
+    # Commit comment
+    def __init__(self):
+        super().__init__(
+            'update',
+            'Update projects descibed in west.yml',
+            _wrap('''
+            Updates all projects according to the manifest file, `west.yml`,
+            in the manifest repository.
+            All projects will be checked out in detached HEAD and the local
+            manifest-rev branch will be reset hard to same HEAD.
+
+            Any local branches will not be altered by this command.
+
+            This command will not touch the manifest repository.
+            '''))
+
+    def do_add_parser(self, parser_adder):
+        return _add_parser(parser_adder, self, _no_update_arg,
+                           _project_list_arg)
+
+    def do_run(self, args, user_args):
+        if args.update:
+            _update_west()
+
+        for project in _projects(args, listed_must_be_cloned=False,
+                                 exclude_manifest=True):
+            branch = _current_branch(project)
+
+            _fetch(project)
+            _checkout_detach(project, _MANIFEST_REV_BRANCH)
+            sha = _sha(project, 'HEAD')
+
+            if branch is not None:
+                _inf(project,
+                     '{name} has been updated and is in detached HEAD.')
+                _inf(project,
+                     'To checkout and rebase \'{}\' with \'{}\', run:'
```

Comment:
Just a nit since I think you're going to need to change this patch, but how about changing "with" to "onto"?

Reply:
yep, onto sounds better.
But then I should also change order, so end result becomes:
`To checkout and rebase {branch} onto {sha (detached-head)}`
agree ?

Reply:
Agree